### PR TITLE
Add sandbox support for nsjail

### PR DESCRIPTION
**Pull request by apmorton**
See #1440

Created a new PR because its on a branch on this repo (so I can try it out on staging)

### CODE REVIEW

**Comment by partouf** [comment]
I can't seem to run this locally, did you configure anything else besides this and the ce-image setup?

I tried it out manually (with -v) after it failed in CE and got the following;

`partouf@ubi2:/tmp/cetest$ nsjail --config /opt/compiler-explorer/ce/etc/nsjail/sandbox.cfg --cwd /app --bindmount /tmp/cetest:/app --env=LD_LIBRARY_PATH=/opt/compiler-explorer/gcc-9.1.0/lib:/opt/compiler-explorer/gcc-9.1.0/lib32:/opt/compiler-explorer/gcc-9.1.0/lib64 -v -- ./a.out
`
image

File: etc/nsjail/sandbox.cfg [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+mount {
+    src: "/opt/compiler-explorer"
+    dst: "/opt/compiler-explorer"
+    is_bind: true
+}
+
+# TODO: review seccomp audit logs
+# this may be better off as a default allow policy with specific syscalled denied
+seccomp_string: "ALLOW {"
```
Comment (mattgodbolt) [inline etc/nsjail/sandbox.cfg]: wowser... we may want to build this cfg in the Makefile using some crappy templating if it ever needs editing :)

**Comment by partouf** [comment]
Remaining things before merging this PR:

* Change to compiler-explorer-image to:
  - Create a new account for Node that doesn't have sudo
  - Start node with this user
* Test these changes in beta (various compilers (note don't forget intel) + some libraries, incl boost) to make sure this user has access to all needed files.

**Comment by mattgodbolt** [comment]
> Remaining things before merging this PR:
> 
> * Change to compiler-explorer-image to:
>   * Create a new account for Node that doesn't have sudo
Done!
>   * Start node with this user
Done!
> * Test these changes in beta (various compilers (note don't forget intel) + some libraries, incl boost) to make sure this user has access to all needed files.
Done!

Does that mean we're ready? :) I think so?

File: etc/nsjail/sandbox.cfg (hunk 1 above)
Comment (mattgodbolt) [reply etc/nsjail/sandbox.cfg:114]: As per slack, will comment this out in a followup change. Thank @apmorton

**Pull request merged**